pennylane/operation.py

```diff
@@ -1187,6 +1196,9 @@ def expand(self):
         Returns:
             .QuantumTape: quantum tape
         """
+        if not self.has_decomposition:
+            raise DecompositionUndefinedError
```

Comment:
Skip creating, entering and exiting a tape if there is no decomposition anyways.